cli/objects/failure_rule.py

```diff
@@ -185,3 +187,15 @@ def _get_ignore(self, rule_dict: dict[Any, Any]) -> bool:
             f'Value for "ignore" is not a boolean or string value in firewatch rule: "{rule_dict}"',
         )
         exit(1)
+
+    def matches_failure(self, failure: "Failure") -> bool:
+        if (
+            hasattr(self, "step")
+            and fnmatch.fnmatch(failure.step, self.step)
```

Comment:
> failure.step and self.step are strings? Do you compare if they are identical?

Yes, they are strings which can be also be in glob format. So "gather-*" would match with "gather-must-gather", for example.

>  Asking since I want to know why fnmatch needed here.

I do not know if there are other reasons `fnmatch` was used in the existing code (I tried to keep refactoring to a minimum in this fix, so most extracted functions were left as-is until review).